### PR TITLE
[BUGFIX] Custom matchers receive the entire option even when used along with searchField

### LINKED ISSUE #473: searchField and matcher incompatibility
Hey,

I have a `power-select-multiple` of people of the form `{name, email}`. If I don't set the `searchField` and hit backspace, I get `Assertion Failed: `{{power-select-multiple}}` requires a 'searchField' when the options are not strings to remove options using backspace`. But when I set `searchField="email"`, then the matcher mis-behaves:

```js
recipientsSearchMatcher(user, term) {
  // I expect user to be the object {name, email}, but it is only the email
}
```

I run `0.10.1`.

Is that an expected behaviour or am I missing something? I would personally that the backspace doesn't throw an exception, it is fine to remove the item altogether without entering edit mode. If that's not an option, I would expect the matcher to still get the object and not just the email.

### PR DESCRIPTION
**Pull request by cibernox**
Closes #473

### CODE REVIEW

**Comment by cibernox** [comment]
@karellm I think that your expectation in the bug report was actually accurate and a better idea that the idea you ended implementing in the PR.

My understanding, as yours in that issue, is that if the user provides a custom matcher to the component, that function should have access to the entire object and the search term. The `searchField` is only convenience to customize the default matcher and has no role here.

I actually consider the decission using the `searchField` for both searching and inferring the text to be displayed in the input when pressing backspace is probably not the best I've taken. Having that property fulfilling two different roles lead to this mess.

Check if this fixes your issue.

File: tests/integration/components/power-select/custom-search-test.js [hunk 1]
```diff
@@ -551,3 +551,18 @@ test('BUGFIX: When the given options are a promise and a search function is prov
   typeInSearch("");
   assert.equal($('.ember-power-select-option').length, 20, 'There is 20 options again§');
 });
+
+test('BUGFIX: If the user provides a custom matcher, the that matcher receives the entire option even if the user also provided a searchField', function(assert) {
```
Comment (karellm) [inline tests/integration/components/power-select/custom-search-test.js]: There is a typo `the that`

Comment (cibernox) [inline tests/integration/components/power-select/custom-search-test.js]: 👀

**Comment by karellm** [comment]
It looks great, should definitely fix my issue. I will implement when merged.

**Pull request merged**